### PR TITLE
Create a runbook to document local stream file verification

### LINKED ISSUE #5799: Create a runbook to document local stream file verification
### Problem

As a mirror node operator, I would like to know the steps to take if asked to verify a modified set of stream files locally.

### Solution

* Add a `docs/runbook/stream-verification.md` that documents how to run the importer locally with the necessary config to verify both S3 and LOCAL providers in case the files are in a bucket or provided directly.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by jnels124**
Add document to describe how to verify stream files locally

**Related issue(s)**:

Fixes #5799

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: docs/runbook/stream-verification.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Stream Verification
+
+## Problem
+Stream files need to be verified for correctness or to identify a particular error observed when processing in an
+environment.
+
+## Setup Database
+
+1. cd {HEDERA_REPO}
```
Comment (steven-sheehy) [inline docs/runbook/stream-verification.md:9]: Could remove HEDERA_REPO everywhere and just put a note at the top like:

> Note: All commands assumed to be ran from the git repository root

File: docs/runbook/stream-verification.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          sidecar: true
+```
+## Run the application
+1. cd {HEDERA_REPO}
+2. `./gradlew importer:bootRun`
+3. Optionally, run in an IDE and run the bootrun gradle task on the importer module
+
+## Verify Data
+1. `psql -U mirror_node`
+2. password is located in docker-compose.yml in this repo and defaulted to `mirror_node_pass`
```
Comment (steven-sheehy) [inline docs/runbook/stream-verification.md:63]: Should add additional verifications like verifying logs for no errors and verify new account balance and record files are showing up in their tables.

File: docs/runbook/stream-verification.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+hedera:
+  mirror:
+    importer:
+      startDate: 2023-04-19T22:47:16.030078003Z # exclusive
+      endDate: 2023-04-19T22:47:16.030078003Z # inclusive
+      network: testnet
+      downloader:
+        cloudProvider: LOCAL
+        mirrorProperties:
+          dataPath: {PATH_TO_RECORD_STREAMS} # will create sub dirs /streams/accountBalances/record{NODE1...NODEn} /streams/recordstreams/record{NODE1...NODEn}. Must have the signature file and the record file for 1/3 of nodes configured in address book
```
Comment (steven-sheehy) [inline docs/runbook/stream-verification.md:51]: Maybe add this comment to the paragraph above.

File: docs/runbook/stream-verification.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+## Problem
+Stream files need to be verified for correctness or to identify a particular error observed when processing in an
+environment.
+
+## Setup Database
+
+1. cd {HEDERA_REPO}
+2. rm -r db
+3. docker-compose up db
```
Comment (steven-sheehy) [inline docs/runbook/stream-verification.md:11]: There might be errors if you don't also run redis. Should either run it or disable it via importer config.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Review by steven-sheehy — DISMISSED** [review]

**Review by MarcKriguerAtHedera — APPROVED** [review]
LGTM

File: docs/runbook/stream-verification.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+1. cd {HEDERA_REPO}
+2. `./gradlew importer:bootRun`
+3. Optionally, run in an IDE and run the bootrun gradle task on the importer module
+
+## Verification
+
+### DB
+
+1. `psql -U mirror_node`
```
Comment (xin-hedera) [inline docs/runbook/stream-verification.md:80]: without it, psql tries to connect through unix socket

```suggestion
1. `psql -h localhost -U mirror_node`
```

File: docs/runbook/stream-verification.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+hedera:
+  mirror:
+    importer:
+      startDate: 2023-04-19T22:47:16.030078003Z # exclusive
+      endDate: 2023-04-19T22:47:16.030078003Z # inclusive
+      network: testnet
+      downloader:
+        cloudProvider: LOCAL
+        mirrorProperties:
+          dataPath: { PATH_TO_RECORD_STREAMS }
```
Comment (xin-hedera) [inline docs/runbook/stream-verification.md:64]: I believe `dataPath` should be at `hedera.mirror.importer.dataPath`

File: docs/runbook/stream-verification.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        secretKey: <S3_SECRET_KEY>
+        bucketName: # Remove or use a known bucket name
+      parser:
+        record:
+          sidecar: true
+```
+
+#### For Local Files application.yml
+
+When starting the application, the following subdirectories will be created under `${hedera.mirror.importer.downloader.mirrorProperties.dataPath}`
```
Comment (xin-hedera) [inline docs/runbook/stream-verification.md:46]:
```suggestion
When starting the application, the following subdirectories will be created under `${hedera.mirror.importer.dataPath}`
```

File: docs/runbook/stream-verification.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+`/streams/accountBalances/record{NODE1...NODEn}` `/streams/recordstreams/record{NODE1...NODEn}` where n is the number of nodes in the address book.
```
Comment (xin-hedera) [inline docs/runbook/stream-verification.md:47]:
```suggestion
`/streams/accountBalances/balance{NODE1...NODEn}` `/streams/recordstreams/record{NODE1...NODEn}` where n is the number of nodes in the address book.
```

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: docs/runbook/stream-verification.md (hunk 6 above)
Comment (jnels124) [reply docs/runbook/stream-verification.md:64]: This sets it specifically for the CommonDownloaderProperties. I can look at the higher level prop to make sure it works set there.

Comment (xin-hedera) [reply docs/runbook/stream-verification.md:64]: It should work and in fact it’s the same bean. Although setting is here also works but we should follow what’s in our configuration.md

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**